### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by steveWDamesJr**
:sunglasses: This is @steveWDamesJr 's 2nd milestone for the OOP-School-Library build. :sunglasses:

- [x] Create class Nameable - returns name attribute
- [x] Create class Decorator - returns the result of the correct_name method @nameable
- [x] Create class CapitalizeDecorator - capitalizes the output of @nameable.correct_name
- [x] Create class TrimmerDecorator - Implements a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it trims the word.
- [x] Run Linters and correct errors

library

Thank you for taking the time to review this PR :pray:

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+require_relative './nameable'
 class Person
   attr_accessor :name, :age
   attr_reader :id
```
Comment (Gambit142) [inline person.rb]: - [x] Great work implementing the requirements for this project. Kindly make the `Person` class inherit from the `Nameable` class based on the project requirements. 👍

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
+  end
```
Comment (Gambit142) [inline decorator.rb]: - [x] The correct_name method for the `Decorator` class should return the `correct_name` method of the `nameable` object it takes in as a parameter during initialization. Kindly implement this change. 👍

File: capitalize_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require_relative './decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameavle.correct_name.capitalize
+  end
```
Comment (Gambit142) [inline capitalize_decorator.rb]: - [x] Kindly fix the typographical error you have here. It should be `@nameable` and not `@nameavle`. 👍

File: trimmer_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require_relative './decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name > 10 ? @nameable.correct_name[0...9].to_s : @nameable.correct_name
+  end
```
Comment (Gambit142) [inline trimmer_decorator.rb]: - [x] Currently, you are comparing a string to `10`. Kindly compare the length of the string to 10. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @steveWDamesJr 👋, 

Good job so far!

## Highlights
- Created `Nameable` class 💯 
- Created `Base` Decorator class 💯
- Created other types of `Decorators` 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

#### I am making this comment here because I can't highlight the piece of code in the **_files changed_** section.

- You can also consider correcting the syntax error of the parameters in the initialize method of the Person class. This is the syntax error `age:,`. Kindly fix it. 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by keroloussamy — APPROVED** [review]
Hi @steveWDamesJr, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**